### ISSUE TITLE
Support Elasticsearch 7

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/SuluArticleBundle#431]
| Q | A
| --- | ---
| Bug? | no
| New Feature? | yes
| SuluArticleBundle Version | for 2.0.0
| Sulu Version | 2.0.3

#### Actual Behavior

Currently we do not support elasticsearch 7

#### Expected Behavior

We should support elasticsearch 7.

#### Steps to Reproduce

`composer.json`

```json
        "elasticsearch/elasticsearch": "7.5.0 as 5.5.0",
        "ongr/elasticsearch-dsl": "7.0.0 as 6.0.4",
        "ongr/elasticsearch-bundle": "^5.2@dev",
```

Then run `bin/console sulu:article:reindex` will end with:

<details>

<summary>Mapping Error</summary>

```
{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "Root mapping definition has unsupported parameters: [article : {properties={authored={type=date}, changer_contact_id={type=text}, structure_type={type=keyword}, parent_page_uuid={type=keyword}, local ization_state={type=object, properties={state={type=keyword}, locale={type=keyword}}}, locale={type=keyword}, title={type=text, fields={raw={type=keyword}, value={type=text}}}, type={type=keyword}, uuid={type=keyword}, type_translation={type=keyword}, creator_contact_i d={type=text}, pages={type=object, properties={page_number={type=integer}, title={type=text, fields={raw={type=keyword}, value={type=text}}}, uuid={type=keyword}, content_data={type=keyword}, route_path={type=text, fields={raw={type=keyword}, value={type=text}}}}}, seo ={type=object, properties={keywords={type=text, fields={raw={type=keyword}, value={type=text}}}, no_index={type=boolean}, no_follow={type=boolean}, hide_in_sitemap={type=boolean}, description={type=text, fields={raw={type=keyword}, value={type=text}}}, title={type=text , fields={raw={type=keyword}, value={type=text}}}, canonical_url={type=text, fields={raw={type=keyword}, value={type=text}}}}}, created={type=date}, teaser_media_id={type=integer}, main_webspace={type=keyword}, additional_webspaces={type=keyword}, author_full_name={typ e=text, fields={raw={type=keyword}, value={type=text}}}, published={type=date}, creator_full_name={type=text, fields={raw={type=keyword}, value={type=text}}}, changer_full_name={type=text, fields={raw={type=keyword}, value={type=text}}}, published_state={type=boolean}, excerpt={type=object, properties={images={type=object, properties={copyright={type=keyword}, formats={type=keyword}, id={type=integer}, title={type=keyword}, url={type=keyword}}}, more={type=text, fields={raw={type=keyword}, value={type=text}}}, icon={type=object, pro perties={copyright={type=keyword}, formats={type=keyword}, id={type=integer}, title={type=keyword}, url={type=keyword}}}, description={type=text, fields={raw={type=keyword}, value={type=text}}}, categories={type=object, properties={keywords={type=text}, name={type=text , fields={raw={type=keyword}, value={type=text}}}, id={type=integer}, key={type=text, fields={raw={type=keyword}, value={type=text}}}}}, title={type=text, fields={raw={type=keyword}, value={type=text}}}, tags={type=object, properties={name={type=text, fields={raw={type =keyword}, value={type=text}}}, id={type=integer}}}}}, author_id={type=text}, teaser_description={type=text}, content_data={type=binary}, route_path={analyzer=pathAnalyzer, type=text, fields={raw={type=keyword}, value={type=text}}}, changed={type=date}}}]"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "Failed to parse mapping [_doc]: Root mapping definition has unsupported parameters: [article : {properties={authored={type=date}, changer_contact_id={type=text}, structure_type={type=keyword}, parent_page_uuid={type=keyword}, localiza tion_state={type=object, properties={state={type=keyword}, locale={type=keyword}}}, locale={type=keyword}, title={type=text, fields={raw={type=keyword}, value={type=text}}}, type={type=keyword}, uuid={type=keyword}, type_translation={type=keyword}, creator_contact_id={ type=text}, pages={type=object, properties={page_number={type=integer}, title={type=text, fields={raw={type=keyword}, value={type=text}}}, uuid={type=keyword}, content_data={type=keyword}, route_path={type=text, fields={raw={type=keyword}, value={type=text}}}}}, seo={t ype=object, properties={keywords={type=text, fields={raw={type=keyword}, value={type=text}}}, no_index={type=boolean}, no_follow={type=boolean}, hide_in_sitemap={type=boolean}, description={type=text, fields={raw={type=keyword}, value={type=text}}}, title={type=text, f ields={raw={type=keyword}, value={type=text}}}, canonical_url={type=text, fields={raw={type=keyword}, value={type=text}}}}}, created={type=date}, teaser_media_id={type=integer}, main_webspace={type=keyword}, additional_webspaces={type=keyword}, author_full_name={type=t ext, fields={raw={type=keyword}, value={type=text}}}, published={type=date}, creator_full_name={type=text, fields={raw={type=keyword}, value={type=text}}}, changer_full_name={type=text, fields={raw={type=keyword}, value={type=text}}}, published_state={type=boolean}, ex cerpt={type=object, properties={images={type=object, properties={copyright={type=keyword}, formats={type=keyword}, id={type=integer}, title={type=keyword}, url={type=keyword}}}, more={type=text, fields={raw={type=keyword}, value={type=text}}}, icon={type=object, proper ties={copyright={type=keyword}, formats={type=keyword}, id={type=integer}, title={type=keyword}, url={type=keyword}}}, description={type=text, fields={raw={type=keyword}, value={type=text}}}, categories={type=object, properties={keywords={type=text}, name={type=text, f ields={raw={type=keyword}, value={type=text}}}, id={type=integer}, key={type=text, fields={raw={type=keyword}, value={type=text}}}}}, title={type=text, fields={raw={type=keyword}, value={type=text}}}, tags={type=object, properties={name={type=text, fields={raw={type=ke yword}, value={type=text}}}, id={type=integer}}}}}, author_id={type=text}, teaser_description={type=text}, content_data={type=binary}, route_path={analyzer=pathAnalyzer, type=text, fields={raw={type=keyword}, value={type=text}}}, changed={type=date}}}]",
    "caused_by": {
      "type": "mapper_parsing_exception",
      "reason": "Root mapping definition has unsupported parameters: [article : {properties={authored={type=date}, changer_contact_id={type=text}, structure_type={type=keyword}, parent_page_uuid={type=keyword}, localization_state={type=object, properties={state={type=keyword}, locale={type=keyword}}}, locale={type=keyword}, title={type=text, fields={raw={type=keyword}, value={type=text}}}, type={type=keyword}, uuid={type=keyword}, type_translation={type=keyword}, creator_contact_id={type=text}, pages={type= object, properties={page_number={type=integer}, title={type=text, fields={raw={type=keyword}, value={type=text}}}, uuid={type=keyword}, content_data={type=keyword}, route_path={type=text, fields={raw={type=keyword}, value={type=text}}}}}, seo={type=object, properties={ keywords={type=text, fields={raw={type=keyword}, value={type=text}}}, no_index={type=boolean}, no_follow={type=boolean}, hide_in_sitemap={type=boolean}, description={type=text, fields={raw={type=keyword}, value={type=text}}}, title={type=text, fields={raw={type=keyword }, value={type=text}}}, canonical_url={type=text, fields={raw={type=keyword}, value={type=text}}}}}, created={type=date}, teaser_media_id={type=integer}, main_webspace={type=keyword}, additional_webspaces={type=keyword}, author_full_name={type=text, fields={raw={type=k eyword}, value={type=text}}}, published={type=date}, creator_full_name={type=text, fields={raw={type=keyword}, value={type=text}}}, changer_full_name={type=text, fields={raw={type=keyword}, value={type=text}}}, published_state={type=boolean}, excerpt={type=object, prop erties={images={type=object, properties={copyright={type=keyword}, formats={type=keyword}, id={type=integer}, title={type=keyword}, url={type=keyword}}}, more={type=text, fields={raw={type=keyword}, value={type=text}}}, icon={type=object, properties={copyright={type=ke yword}, formats={type=keyword}, id={type=integer}, title={type=keyword}, url={type=keyword}}}, description={type=text, fields={raw={type=keyword}, value={type=text}}}, categories={type=object, properties={keywords={type=text}, name={type=text, fields={raw={type=keyword }, value={type=text}}}, id={type=integer}, key={type=text, fields={raw={type=keyword}, value={type=text}}}}}, title={type=text, fields={raw={type=keyword}, value={type=text}}}, tags={type=object, properties={name={type=text, fields={raw={type=keyword}, value={type=text }}}, id={type=integer}}}}}, author_id={type=text}, teaser_description={type=text}, content_data={type=binary}, route_path={analyzer=pathAnalyzer, type=text, fields={raw={type=keyword}, value={type=text}}}, changed={type=date}}}]"
    }
  },
  "status": 400
}
```

</details>

#### Possible Solutions

I think we need to fork ongr/elasticsearch-bundle:5.2 and update the mapping. Or try to upgrade to elasticsearchbundle 6/7 which need many bc breaks.

**Comment by almare** [comment]
This would be great. Currently my installation of article-bundle in sulu2.0 fails due to the elasticsearch part. I think my previous opened ticket is related to the elasticsearch version see [here](https://github.com/sulu/SuluArticleBundle/issues/409) . Maybe this could be closed after the upgrade.

**Comment by alexander-schranz** [comment]
I had a look what would be needed to support ES7. The change seems to be small but is in the ongr elasticsearch bundle. https://github.com/ongr-io/ElasticsearchBundle/pull/909.

For the one who want to test this need todo the following in its composer.json:

```

    "repositories": [
        {
            "type": "vcs",
            "url": "git@github.com:alexander-schranz/ElasticsearchBundle.git"
        }
    ],
    "require": {
        "elasticsearch/elasticsearch": "^7.5",
        "ongr/elasticsearch-dsl": "^7.0",
        "ongr/elasticsearch-bundle": "dev-alexander-schranz:feature/support-es7-in-version-5",
        "sulu/article-bundle": "^2.0@dev"
```

**Comment by almare** [comment]
@alexander-schranz many thanks for spending your time on this topic.

**Comment by alexander-schranz** [comment]
Merged in #433
